### ISSUE TITLE
Bug in grdimage @earth_day_10m -I%SHADOW%

### ISSUE THREAD

**Issue by Esteban82** [GenericMappingTools/gmt#4003]
There is an error when I plot the blue marble with a shadow effect as you can see in upper figure. When I only plot the blue marble or the relief it is ok. 

```
	gmt grdgradient @earth_relief_10m -Nt1.00 -A45 -G%SHADOW%
	gmt grdimage -R -J -O -K >> %OUT% @earth_day_10m -I%SHADOW% 
```

![BlueMarble+Intes_Bug](https://user-images.githubusercontent.com/17908951/90711390-fb65ea00-e276-11ea-905e-1876e4cd7b81.png)

[Full script](https://www.dropbox.com/s/gli8xkbrcipb6d8/BlueMarble%2BIntes_Bug.bat?dl=0)

Win 64 GMT 6.1

**Comment by PaulWessel** [comment]
I can reproduce it.  Looks like grdimage is not dealing properly with the subset when dataset is an image to be shaded.

**Comment by Esteban82** [comment]
I managed to get the right figure trough gmt grdgrdal (from the post of the forum on "[How to cut an image](https://forum.generic-mapping-tools.org/t/how-to-cut-an-image-geotiff/1831)").

The key commands is
`	gmt grdgdal @earth_day_10m -Gmarble.tif -Atranslate -M+r -F"-projwin -85 26 -54 9"`

I try to use the following instead but I get the original wrong figure.

`gmt grdgdal @earth_day_10m -Gmarble.tif -Atranslate -M+r -R-85/-54/9/26 -I10m`

![Image_Cut](https://user-images.githubusercontent.com/17908951/123574435-32196480-d7a6-11eb-8c51-87bcc7047042.png)

Maybe this could help fix this bug or to find a way around. 

Full script:
```
gmt begin Image_Cut png
	gmt basemap -R-85/-54/9/26 -JM15c -Baf
	gmt grdgradient @earth_relief_10m -Nt0.5 -A45 -Gtemp_intens.nc
	gmt grdgdal @earth_day_10m -Gmarble.tif -Atranslate -M+r -F"-projwin -85 26 -54 9"
	gmt grdimage marble.tif -I"temp_intens.nc"
gmt end show
```

**Comment by meghanrjones** [comment]
Fixed by #5798
